recipes/openssl/1.x.x/conanfile.py

```diff
@@ -223,35 +216,61 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
-
-    def layout(self):
-        pass
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+        self.conf.define("tools.gnu:make_program", self._make_program)
 
     def requirements(self):
-        if self._full_version < "1.1.0" and self.options.get_safe("no_zlib") == False:
```

Comment:
I wonder what assumpt the old code was working with by remove this option https://github.com/openssl/openssl/blob/OpenSSL_1_1_1s/Configure#L475

Seems like it could have been enabled (so I think this option means something bizarre)

For me to test this use case locally

Reply:
Hi @prince-chrismc,
From what I understand, old versions of openssl supported zlib-based compression. It was later determined to be a security vulnerability (e.g., https://www.cvedetails.com/bugtraq-bid/31692/OpenSSL-zlib-Compression-Memory-Leak-Remote-Denial-of-Serv.html) and hence support was dropped (see https://github.com/openssl/openssl/issues/14506). I'd personally opt for removing all support for versions less than 1.1.1 as use of older versions is unsupported and strongly discouraged (see https://www.openssl.org/source/). If CCI is okay with dropping support for those obsolete releases, I would happily rip out support for them from this recipe as it will make it much simpler.

Reply:
Sadly there's just so much demand for them. It's like old version of boost 🤷 They become a nightmare to maintain 🙊 

Thanks a ton for that extra info!